flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/nodes/physical/stream/StreamExecSort.scala

```diff
@@ -38,16 +38,22 @@ import org.apache.calcite.rex.RexNode
   * <p>''SELECT * FROM TABLE ORDER BY ROWTIME, a'' will be converted to [[StreamExecTemporalSort]]
   * <p>''SELECT * FROM TABLE ORDER BY a LIMIT 2'' will be converted to [[StreamExecRank]]
```

Comment:
The basic strategy for selecting a sort/rank operator is a query will be converted to `StreamExecTemporalSort`, `StreamExecFirstLastRow` or `StreamExecRank`  as much as possible, otherwise it will be converted to `StreamExecSort`. Currently, `StreamExecSort` is only used for testing with bounded source. So, in product environment, `StreamExecSort` is disabled, and an exception will be thrown if a query is converted to `StreamExecSort`.

Reply:
This kind of ambiguous is we should try to avoid, it's the meaning to have SQL. How can a user or a developer to learn the differences between these two operators. You only said in case A, we should use X. In case B, we should use Y. What's the rationale behind these choices?

Reply:
So far as we know, Sort without limit is meaningless for unbounded source. 
In a simple word, Sort with limit will be converted to `StreamExecRank`, Sort without limit will be converted to `StreamExecSort`.